extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -506,7 +507,8 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 		let isPythonInstalled = JupyterServerInstallation.isPythonInstalled();
 
 		// If the latest version of ADS-Python is not installed, then prompt the user to upgrade
-		if (isPythonInstalled && !this._usingExistingPython && utils.compareVersions(await this.getInstalledPythonVersion(this._pythonExecutable), constants.pythonVersion) < 0) {
+		if (!this._upgradePrompted && isPythonInstalled && !this._usingExistingPython && utils.compareVersions(await this.getInstalledPythonVersion(this._pythonExecutable), constants.pythonVersion) < 0) {
```

Comment:
When else is this called - is there possibly an issue here where someone dismisses it and then later on tries to start up a server at which point this never prompts?

Reply:
It won't prompt again in the scenario you described above but the prompt will show again when ADS is restarted.

Reply:
That doesn't seem like the best behavior, right? We should at least be asking the user if they want to dismiss and not show again - but just always hiding the prompt after the first one means things will just silently fail to work properly.

Reply:
Hm I was thinking about the scenario where a user's jupyter server shuts down because they had closed their python notebook and were idle for a while (default 5 minutes). Then opening another python notebook will start a new server - I'm not sure if it would be best to show the prompt again here.

Reply:
I don't think this would be an issue since users can still use Python 3.6.6 without any issues - nothing would fail if they continue to use it. 

@chlafreniere @corivera thoughts here?

Reply:
I take closing the prompt as "don't touch anything for now and leave me alone" and since the user didn't explicitly click "Don't Show Again", it's fine to re-prompt them when ADS starts next time. Agreed that it's more intuitive than every time that a Jupyter server starts up (since that's really an implementation detail).

I think this is fine as coded for now.